python/fbprophet/diagnostics.py

```diff
@@ -127,11 +138,39 @@ def cross_validation(model, horizon, period=None, initial=None, multiprocess=Fal
             msg += 'Consider increasing initial.'
             logger.warning(msg)
 
-    if multiprocess is True:
-        with Pool() as pool:
-            logger.info('Running cross validation in multiprocessing mode')
-            input_df = ((df, model, cutoff, horizon, predict_columns) for cutoff in cutoffs)
-            predicts = pool.starmap(single_cutoff_forecast, input_df)
+    if parallel:
+        if parallel not in {"threads", "processes", "dask"}:
+            raise ValueError("'parallel' must be one of 'threads', 'processes',"
+                             " or 'dask'. Got '{}' instead".format(parallel))
+
+        if parallel == "threads":
+            pool = concurrent.futures.ThreadPoolExecutor()
+        elif parallel == "processes":
+            pool = concurrent.futures.ProcessPoolExecutor()
+        elif parallel == "dask":
+            try:
+                import dask
+                from dask.distributed import get_client
+            except ImportError:
+                raise ImportError("parallel='dask' requies the optional "
+                                  "dependency dask.")
+            pool = get_client()
+            # delay df and model to avoid large objects in task graph.
+            df, model = pool.scatter([df, model])
```

Comment:
These objects are a bit large. Doing this pre-scattering helps improve the performance for Dask, and avoids a warning bubbling up to the user about large objects in the task graph.